From 1d27a7ef452cdfcc7dd6796102bae4b1d36cef2c Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Tue, 26 Mar 2019 21:52:24 +0100
Subject: [PATCH] Simplify logging to console (stdout)

Systemd-friendly format without timestamp.

Signed-off-by: Edmunt Pienkowsky <roed@onet.eu>

diff --git a/lib/carbon/log.py b/lib/carbon/log.py
index cdb53ed..ab95b75 100644
--- a/lib/carbon/log.py
+++ b/lib/carbon/log.py
@@ -111,7 +111,7 @@ class CarbonLogObserver(object):
 
   @staticmethod
   def stdout_observer(event):
-    stdout.write(formatEvent(event, includeType=True) + '\n')
+    stdout.write(formatEventSimple(event) + '\n')
     stdout.flush()
 
   def logdir_observer(self, event):
@@ -131,18 +131,16 @@ class CarbonLogObserver(object):
 
 carbonLogObserver = CarbonLogObserver()
 
+def formatEventSimple(event):
+  event['isError'] = 'failure' in event
+  return textFromEventDict(event)
 
-def formatEvent(event, includeType=False):
+def formatEvent(event):
   event['isError'] = 'failure' in event
   message = textFromEventDict(event)
 
-  if includeType:
-    typeTag = '[%s] ' % event.get('type', 'console')
-  else:
-    typeTag = ''
-
   timestamp = time.strftime("%d/%m/%Y %H:%M:%S")
-  return "%s :: %s%s" % (timestamp, typeTag, message)
+  return "%s :: %s" % (timestamp, message)
 
 
 logToDir = carbonLogObserver.log_to_dir
-- 
2.21.0

