From d4ab95d4f0ca9e3bac52854f7492243f44a0e025 Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <reod@onet.eu>
Date: Tue, 26 Mar 2019 21:05:19 +0100
Subject: [PATCH 3/4] util.logtime - raise log level from info to debug

Signed-off-by: Edmunt Pienkowsky <reod@onet.eu>
Signed-off-by: Edmunt Pienkowsky <roed@onet.eu>

diff --git a/webapp/graphite/util.py b/webapp/graphite/util.py
index 1b620331..b27bbdb2 100644
--- a/webapp/graphite/util.py
+++ b/webapp/graphite/util.py
@@ -260,7 +260,7 @@ class Timer(object):
     self.name = name
 
   def stop(self):
-    log.info(
+    log.debug(
       '{name} :: {msg} {sec:.6}s'.format(
         name=self.name,
         msg=self.msg,
