From 87ab4acda8d8ef3807d22d225ca185b8b48c6d26 Mon Sep 17 00:00:00 2001
From: Edmunt Pienkowsky <roed@onet.eu>
Date: Fri, 10 Jan 2020 12:33:34 +0100
Subject: [PATCH] Use python_2_unicode_compatible from six module

django.utils.six was removed in Django 3.0.

diff --git a/tagging/models.py b/tagging/models.py
index ea77837..687c35e 100644
--- a/tagging/models.py
+++ b/tagging/models.py
@@ -4,7 +4,6 @@ Models and managers for tagging.
 from django.db import models
 from django.db import connection
 from django.utils.encoding import smart_text
-from django.utils.encoding import python_2_unicode_compatible
 from django.utils.translation import ugettext_lazy as _
 from django.contrib.contenttypes.models import ContentType
 from django.contrib.contenttypes.fields import GenericForeignKey
@@ -16,6 +15,8 @@ from tagging.utils import calculate_cloud
 from tagging.utils import parse_tag_input
 from tagging.utils import get_queryset_and_model
 
+from six import python_2_unicode_compatible
+
 
 qn = connection.ops.quote_name
 
diff --git a/tagging/utils.py b/tagging/utils.py
index cd7e0ee..13ee881 100644
--- a/tagging/utils.py
+++ b/tagging/utils.py
@@ -3,8 +3,8 @@ Tagging utilities - from user tag input parsing to tag cloud
 calculation.
 """
 import math
+import six
 
-from django.utils import six
 from django.db.models.query import QuerySet
 from django.utils.encoding import force_text
 from django.utils.translation import ugettext as _
-- 
2.24.1

